### PR TITLE
ARTEMIS-4252 Fix flaky QuorumFailOverTest tests

### PR DESCRIPTION
**Pull request by brusdev**

### CODE REVIEW

File: artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java [hunk 1]
```diff
@@ -1328,6 +1330,17 @@ protected static final void waitForComponent(final ActiveMQComponent component,
       }
    }
 
+   protected static final void waitForBridges(final ActiveMQServer server, int connectedBridges) throws Exception {
+      waitForBridges(server.getClusterManager().getDefaultConnection(null), connectedBridges);
+   }
+
+   protected static final void waitForBridges(ClusterConnection clusterConnection, int connectedBridges) throws Exception {
+      Bridge[] bridges = clusterConnection.getBridges();
```
Comment (gemmellr) [inline artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java:1338]: This doesnt look used. Is it needed? Was a quick-exit check intended but missed?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/ClusterConnectionImpl.java [hunk 2]
```diff
@@ -630,6 +630,11 @@ public long getCallTimeout() {
       return callTimeout;
    }
 
+   @Override
+   public Bridge[] getBridges() {
+      return records.values().stream().map(MessageFlowRecord::getBridge).toArray(Bridge[]::new);
```
Comment (gemmellr) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/ClusterConnectionImpl.java:635]: The 'records' field has 'recordsGuard' to protect its use in other cases. Is that needed here?

Comment (brusdev) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/ClusterConnectionImpl.java:635]: It makes sense, I'll fix it. I'll create another PR to fix it.

File: artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java (hunk 1 above)
Comment (brusdev) [reply artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java:1338]: You are right I missed to remove that line, I'll create another PR to fix it.

**Pull request merged**